### PR TITLE
bpo-44925: [docs] Confusing deprecation notice for typing.IO

### PR DESCRIPTION
**Pull request by DonnaDia**
https://bugs.python.org/issue44925

### CODE REVIEW

File: Doc/library/typing.rst [hunk 1]
```diff
@@ -1507,8 +1507,7 @@ Other concrete types
    :func:`open`.
 
    .. deprecated-removed:: 3.8 3.12
-      These types are also in the ``typing.io`` namespace, which was
-      never supported by type checkers and will be removed.
+      ``typing.io`` namespace is deprecated and will be removed.
```
Comment (gvanrossum) [inline Doc/library/typing.rst]:
```suggestion
      The ``typing.io`` namespace is deprecated and will be removed.
```

**Comment by Fidget-Spinner** [comment]
There's also the label for ``typing.re`` that needs fixing at https://github.com/python/cpython/pull/28004/files#diff-8a0f115fde6769c122b771b6d0eca184c4580f7b5fabe2f0b0579c679424364fR1522 .

File: Doc/library/typing.rst (hunk 1 above)
Comment (DonnaDia) [reply Doc/library/typing.rst]: This was added to the new commit ✅

File: Doc/library/typing.rst [hunk 2]
```diff
@@ -1507,8 +1507,8 @@ Other concrete types
    :func:`open`.
 
    .. deprecated-removed:: 3.8 3.12
-      These types are also in the ``typing.io`` namespace, which was
-      never supported by type checkers and will be removed.
+      The ``typing.io`` and ``typing.re`` namespaces are deprecated and will be
+      removed.
```
Comment (Fidget-Spinner) [inline Doc/library/typing.rst]: Please also add where people can still get these types from
```suggestion
      removed.  These types should be directly imported from ``typing`` instead.
```

**Review by Fidget-Spinner — CHANGES_REQUESTED** [review]
@DonnaDia sorry if my previous message wasn't clear, but there's another instance of the same deprecation notice below at line 1523 (for `typing.Pattern` and `typing.Match`). Please apply the same changes you did for the current `typing.IO` deprecation notice. Thank you.

File: Doc/library/typing.rst (hunk 2 above)
Comment (DonnaDia) [reply Doc/library/typing.rst]: Thanks for this note!

Comment (DonnaDia) [reply Doc/library/typing.rst]: I have made the requested changes; please review again

**Comment by DonnaDia** [comment]
I have made the requested changes; please review again

**Review by Fidget-Spinner — APPROVED** [review]
Thanks for the contribution! LGTM.

**Comment by Fidget-Spinner** [comment]
@gvanrossum or @srittau, if either of you are unsatisfied, please don't hesitate to reopen the issue or comment on this PR.

@DonnaDia congrats on your first commit to CPython 🥳  !!

**Comment by srittau** [comment]
Looks good to me.

**Pull request merged**